src/main/src/main/java/com/shevtsov/dao/impl/ClientDaoImpl.java

```diff
@@ -0,0 +1,71 @@
+package com.shevtsov.dao.impl;
+
+import com.shevtsov.dao.ClientDao;
+import com.shevtsov.domain.Client;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ClientDaoImpl implements ClientDao {
+    private Map<Long, Client> map = new HashMap<>();        //storage emulation
+    private static long generator = 0;
+    private static ClientDao instance;
```

Comment:
For Singleton it is better make this final

Reply:
done